pkg

```diff
@@ -1,22 +1,38 @@
 # OpenIndiana pkg(5) completion                             -*- shell-script -*-
 # ------------------------------------------------------------------------------
-# Aurelien Larcher <aurelien@openindiana.org>
+# Copyright (c) 2017, Aurelien Larcher <aurelien@openindiana.org>
+# Copyright (c) 2018, Michal Nowak <mnowak@startmail.com>
 
 # Bash-completion does not handle custom wordbreaks so comma-separated values
 # cannot be implemented without this.
 COMP_WORDBREAKS="${COMP_WORDBREAKS//,},"
 
+_get_installed_packages()
+{
+  # XXX: Lists only last part of package's FMRI, path walker is warranted
+  COMPREPLY=( $(compgen -W "$(pkg list -H $* | awk '{print $1}' | awk -F/ '{print $NF}')" -- "${cur}") )
+}
+
+_get_values()
+{
+  local command="$1"
+  COMPREPLY=( $(compgen -W "$(pkg $command -H | awk '{print $1}')" -- "${cur}") )
+}
+
+_get_alternative_output_format()
+{
+  COMPREPLY=( $(compgen -W "tsv json json-formatted" -- "${cur}") )
+}
+
 _pkg5()
 {
   local cur prev words cword
   _init_completion || return
 
   local cmds="list search info contents update install uninstall exact-install history
-              verify fix
-              publisher set-publisher unset-publisher
+              verify fix apply-hot-fix publisher set-publisher unset-publisher update-format
```

Comment:
I originally listed the cmds according to the output of pkg help for easier listing.

Reply:
OK, thanks, now I keep the list as is, only adding missing commands where they likely belong.